### DISCUSSION TITLE
add_mesh_slice origin bug

### DISCUSSION THREAD

**Discussion by rabiislimane** [pyvista/pyvista#2797] [Q&A] [answered]
Hello,

For example if you have two cubes (side to side) seperated by a small gap and you use add_mesh_slice (to add a slicer between the cubes) you will get **ValueError: Empty meshes cannot be plotted. Input mesh has zero points.**
This is because add_mesh_slice uses by default mesh.center position as origin for the plane widget.

I also noticed that this bug only appeared on initialization and did not appear if the slicer was: 1- initialized correctly (with a mesh in the center) 2- moved to an empty location 3- then back to the mesh.

So my question is : 
- Is there any way to specifie the origin when using plotter.add_mesh_slice ? 
- Is there another way to avoid this bug ?

**Comment by banesullivan** [comment, marked as answer]
Captured in https://github.com/pyvista/pyvista/issues/2815 and fixed in https://github.com/pyvista/pyvista/pull/2883
